src/github.com/travis-ci/jupiter-brain/vsphere.go

```diff
@@ -264,6 +280,16 @@ func (i *vSphereInstanceManager) createClient(ctx context.Context) (*govmomi.Cli
 	}
 
 	client.Client.RoundTripper = vim25.Retry(client.Client.RoundTripper, vim25.TemporaryNetworkError(3))
+	client.Client.RoundTripper = &soapBreakerRoundTripper{
+		rt: client.Client.RoundTripper,
+		cb: gobreaker.NewCircuitBreaker(gobreaker.Settings{
+			Name: "vSphere govmomi",
+			ReadyToTrip: func(counts gobreaker.Counts) bool {
+				failureRatio := float64(counts.TotalFailures) / float64(counts.Requests)
+				return counts.Requests >= 3 && failureRatio >= 0.6
```

Comment:
We can do that, but my concern with this is the code and math itself, not so much the numerical constants. I agree that whatever code we end up with, we can probably extract the constants to configs.